core/node/node_framework/src/implementations/layers/eth_sender/aggregator.rs

```diff
@@ -95,7 +102,44 @@ impl WiringLayer for EthTxAggregatorLayer {
     }
 
     async fn wire(self, input: Self::Input) -> Result<Self::Output, WiringError> {
+        tracing::info!(
+            "Wiring tx_aggregator in {:?} mode which is {}",
+            self.settlement_mode,
+            self.settlement_mode.is_gateway()
+        );
+        tracing::info!("Contracts: {:?}", self.contracts_config);
+        tracing::info!("Gateway contracts: {:?}", self.gateway_contracts_config);
         // Get resources.
+
+        let (validator_timelock_addr, multicall3_addr, diamond_proxy_addr) =
+            if self.settlement_mode.is_gateway() {
+                (
+                    self.gateway_contracts_config
+                        .clone()
```

Comment:
`clone()`s here and below are redundant as we've discussed. I'd advise to define a local var as `self.gateway_contracts_config.as_ref().unwrap()` and maybe add error handling.